content/learn/book/welcome/plugins/_index.md

```diff
@@ -0,0 +1,99 @@
++++
+title = "Modularity through plugins"
+weight = 3
+template = "book-section.html"
+page_template = "book-section.html"
++++
+
+In Bevy, virtually all of the functionality you might want is added via **plugins**, which are added to your app.
+These might provide basic windowing functionality, handle input or sound, calculate physics or provide game-specific logic like a tile map system.
+
+By combining Bevy's first-party plugins with third-party alternatives, you can modularly customize the behaviour of the Bevy game engine, and then add your own plugins to contain your own game-specific code in a well-organized way.
+
+## Writing your own plugins
+
+Plugins are collections of code that modify the `App` (which controls all of the data and logic of our game) using the builder pattern.
+Any code in a plugin could be directly substituted directly on the base `App`.
+There's no magic to be found here; they're just a straightforward tool for code organization.
+
+You can write your own to organize your own code by implementing the `Plugin` trait on a struct of your own creation.
+
+```rust
+use bevy::prelude::*;
+
+fn main(){
+ App::build()
+  // As discussed below, DefaultPlugins provide the standard scaffolding for Bevy games
+   .add_plugins(DefaultPlugins)
+   // Plugins merely organize code: this could just be replaced directly with 
+   // the init_resource and add_system calls from below
+   .add_plugin(ScorePlugin)
+   .run();
+}
+
+struct ScorePlugin;
+
+impl Plugin for ScorePlugin {
+  fn build(&self, app: &mut App) {
+     app
+       // The Score struct is addded as a resource (global singleton) to the world, 
+       // beginning at the default value of 0
+       .init_resource::<Score>()
+       // Increments the score by 1 every pass of the game loop
+       .add_system(increment_score.system())
+       // Prints the current value of the score
+       .add_system(report_score.system());
+  }
+}
+
+#[derive(Default, Debug)]
+struct Score(u8);
+
+fn increment_score(score: ResMut<Score>){
+  score.0 += 1;
+}
+
+fn report_score(score: Res<Score>){
+  info!(score);
+}
+```
+
+## `DefaultPlugins`
+
+Bevy's [`DefaultPlugins`](https://docs.rs/bevy/latest/bevy/struct.DefaultPlugins.html) are intended to get up and running with a "standard game" as quickly as possible. Let's take a look at the [source](https://github.com/bevyengine/bevy/blob/latest/crates/bevy_internal/src/default_plugins.rs) to see what's included.
```

Comment:
Any particular reason for using manual links instead of the short codes? They save space, cut down on redundancy, and ensure consistency:

```md
{{rust_type(type="struct" crate="bevy_core", name="CorePlugin")}}
[`CorePlugin`](https://docs.rs/bevy/latest/bevy/core/struct.CorePlugin.html)

{{rust_type(type="struct" crate="bevy_app", name="AppBuilder" method="add_system")}}
[`AppBuilder::add_system()`](https://docs.rs/bevy_app/latest/bevy_app/struct.AppBuilder.html#method.add_system)
```

Reply:
The "structured" approach also opens the doors to smarter formatting in the future (ex: different colors for traits vs structs)

Reply:
I just didn't know about the short codes :P

I'll do a cleanup pass on all of them...

Reply:
Awesome. Thanks!